types/index.d.ts

```diff
@@ -1,226 +1,218 @@
-     *
-     * If asynchronous plugins are configured an error is thrown.
-     *
-     * @param file
-     * @returns Virtual file with modified contents.
-     */
-    processSync(file: VFileCompatible): vfile.VFile
-
-    /**
-     * Get or set information in an in-memory key-value store accessible to all phases of the process.
-     * An example is a list of HTML elements which are self-closing, which is needed when parsing, transforming, and compiling HTML.
-     *
-     * @returns key-value store object
-     */
-    data(): {[key: string]: unknown}
-    /**
-     * @param key Identifier
-     * @returns If getting, the value at key
-     */
-    data(key: string): unknown
-    /**
-     * @param value Value to set. Omit if getting key
-     * @returns If setting, the processor on which data is invoked
-     */
-    data(key: string, value: any): Processor
-
-    /**
-     * Freeze a processor. Frozen processors are meant to be extended and not to be configured or processed directly.
-     *
-     * Once a processor is frozen it cannot be unfrozen. New processors functioning just like it can be created by invoking the processor.
-     *
-     * It’s possible to freeze processors explicitly, by calling `.freeze()`, but `.parse()`, `.run()`, `.stringify()`, and `.process()` call `.freeze()` to freeze a processor too.
-     *
-     * @returns The processor on which freeze is invoked.
-     */
-    freeze(): Processor
-  }
-
-  type Plugin = Attacher
-  type Settings = {
-    [key: string]: unknown
-  }
-  /**
-   * Presets provide a potentially sharable way to configure processors.
-   * They can contain multiple plugins and optionally settings as well.
-   */
-  interface Preset {
-    plugins?: PluggableList
-    settings?: Settings
-  }
-  type PluginTuple = [Plugin, Settings]
-  type Pluggable = Plugin | Preset | PluginTuple
-  type PluggableList = Pluggable[]
-
-  /**
-   * An attacher is the thing passed to `use`.
-   * It configures the processor and in turn can receive options.
-   *
-   * Attachers can configure processors, such as by interacting with parsers and compilers, linking them to other processors, or by specifying how the syntax tree is handled.
-   *
-   * @this Processor context object is set to the invoked on processor.
-   * @param options Configuration
-   * @returns Optional.
-   */
-  interface Attacher {
-    (this: Processor, options?: unknown): Transformer | void
-  }
-
-  /**
-   * Transformers modify the syntax tree or metadata of a file. A transformer is a function which is invoked each time a file is passed through the transform phase.
-   * If an error occurs (either because it’s thrown, returned, rejected, or passed to `next`), the process stops.
-   *
-   * The transformation process in unified is handled by `trough`, see it’s documentation for the exact semantics of transformers.
+import {Node} from 'unist'
+import vfile, {VFile, VFileOptions, VFileContents} from 'vfile'
```

Comment:
All of the modules in unified ecosystem are using cjs. We should import it like `import something = require('something')`. https://github.com/Microsoft/dtslint/blob/master/docs/no-import-default-of-export-equals.md

So we have two choices.

1. Retrieve types like `vfile.*`.

```ts
import vfile = require('vfile')

// Now we can use the interfaces like `vfile.VFile`
```

2. Have 2 import statements.

```ts
import vfile = require('vfile')
import {VFile, VFileOptions, VFileContents} from 'vfile'
```

Reply:
@Rokt33r theres the conundrum.
the
```ts
import vfile = require('vfile')
```

style imports are part of what is causing CI on master branch to fail.
See the logs in #46.

Reply:
I found that the linter is overly critical. We could turn off some of the linting maybe?

Reply:
@Rokt33r 

It works fine for me. I submitted a pr to this pr. https://github.com/ChristianMurphy/unified/pull/1

It might be dependency problems.

```
unified@7.1.0 /Users/junyoung/Code/unified
├─┬ dtslint@0.4.9
│ └── typescript@3.6.0-dev.20190608 
└── typescript@3.5.1
```

If you are still getting an error, could you share the error log?

Reply:
> I found that the linter is overly critical. We could turn off some of the linting maybe?

@wooorm We can do by modifying `types/tslint.json`.

Reply:
merged, thanks @Rokt33r!